### ISSUE TITLE
User Gstreamer pipelines with parameters fail to run

### ISSUE THREAD

**Issue by izaera** [quodlibet/quodlibet#3828]
Steps to reproduce
------------------
I have a Gstreamer pipeline to upsample CD audio to 88.2KHz that works well with `get-launch-1.0` if I place it after `uridecodebin uri="mysong.flac"`. However, when I paste it into quodlibet it doesn't work.

The pipeline is the following: `audioconvert ! audioresample ! audio/x-raw,rate=88200 ! alsasink device=hw:2,0`

And I suspect the culprit is the `! audio/x-raw,rate=88200` bit, which is a parameter to `audioresample` because I get the following error when running quodlibet with --debug:

```
D: 07.037: player.gstbe.player.__init_pipeline: User pipeline (from player.gst_pipeline): 'audioconvert ! audioresample ! audio/x-raw,rate=88200 ! alsasink device=hw:2,0'
0:13:26.877778890 1193359 0x55969bc25e60 ERROR           GST_PIPELINE gst/parse/grammar.y:857:priv_gst_parse_yyparse: no element "audio"

(io.github.quodlibet.QuodLibet:1193359): GStreamer-CRITICAL **: 13:25:49.998: gst_element_make_from_uri: assertion 'gst_uri_is_valid (uri)' failed
0:13:26.877853008 1193359 0x55969bc25e60 ERROR           GST_PIPELINE gst/parse/grammar.y:1014:priv_gst_parse_yyparse: no source element for URI "/x-raw,rate=88200"
```

I believe the problem is in this [part of the code](https://github.com/quodlibet/quodlibet/blob/97f2e65788c53df7215710b8a5b3e5275ad6c3aa/quodlibet/player/gstbe/util.py#L164), but since I'm no expert in Python, nor in Gstreamer, I'm not sure if I'm right.

Expected Output
---------------
It should work.

Actual Output
-------------
Doesn't work :sweat_smile: 

Test System
-----------

### Which version of Quod Libet?
4.4.0

### Which operating system 
ArchLinux (uname -a: Linux baco 5.14.11-arch1-1 #1 SMP PREEMPT Sun, 10 Oct 2021 00:48:26 +0000 x86_64 GNU/Linux)

### If it's audio-related, what back-end?
gstreamer 1.18.5

Additional Information
----------------------
Nothing else. I just need guidance on how to configure the pipeline properly or confirmation that this is a bug.

**Comment by declension** [comment]
Yes, I think you're right. We should probably use `gst_parse_launch` here rather than simplistic splitting.

But then this whole thing could do with rewriting - see #3512, so we'd do well to keep complex custom pipelines in mind (assuming they continue to be sorted)

**Comment by izaera** [comment]
So, what's the proposition? Close this issue and wait for the other to happen? Or maybe send a small fix for this based on `gst_parse_launch` while the other arrives? :thinking: 

I can give it a try (even if I have no idea of how GStreamer works and the last time I did something with Python was years ago). But hey, looks like a good place to start :sweat_smile:

**Comment by declension** [comment]
I can't think of any easy fixes, but yes that would be good. The #3512 will probably take ages (currently).

**Comment by izaera** [comment]
OK :+1: I'll give it a try when I have time, then. Thx :-)

**Comment by nagisa** [comment]
`audio/x-raw,rate=88200` when passed to `gst-launch-1.0` is a shortcut for `capsfilter caps=audio/x-raw,rate=88200` which in my experience doesn't (always?) work with `gst_parse_launch` either, so changing quodlibet's code might not actually solve anything here.
